tests/src/services/currencyAdapter.service.test.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  AllowedCryptoCurrencies,
+  AllowedFiatCurrencies,
+  CurrencyAdapter,
+} from '../../../src/services/currencyAdapter.service';
+
+describe('Currency Adapter Service', () => {
+  describe('Normalize currency for Stripe', () => {
+    test('When a valid fiat currency is provided, then it returns lowercase format', () => {
+      const result = CurrencyAdapter.normalizeForStripe('EUR');
+
+      expect(result).toBe('eur');
+    });
+
+    test('When a valid fiat currency in mixed case is provided, then it returns lowercase format', () => {
+      const result = CurrencyAdapter.normalizeForStripe('UsD');
+
+      expect(result).toBe('usd');
+    });
+
+    test('When a valid cryptocurrency is provided, then it returns fallback fiat currency that is eur in lowercase format', () => {
+      const result = CurrencyAdapter.normalizeForStripe('BTC');
+
+      expect(result).toBe('eur');
+    });
+
+    test('When an unsupported currency is provided, then it throws an error', () => {
+      expect(() => CurrencyAdapter.normalizeForStripe('INVALID')).toThrow('Currency INVALID is not supported');
```

Comment:
If you can create a custom error and check the instaceof it, better, as that makes the test easier to maintain, otherwise, changing the error message will require a change here

Reply:
I can use the generic `BadRequestError` to check what error was thrown instead of the message.